assets/src/dashboard/components/pill/components.js

```diff
@@ -0,0 +1,163 @@
+ */
+import { ReactComponent as CheckmarkIcon } from '../../icons/checkmark.svg';
+import { ReactComponent as CloseIcon } from '../../icons/close.svg';
+import { BEZIER } from '../../constants';
+
+export const ACTIVE_CHOICE_ICON_SIZE = 16;
+export const ACTIVE_CHOICE_LEFT_MARGIN = 4;
+
+export const PillContainer = styled.label`
+  ${({ theme }) => `
+    display: inline-flex;
+    justify-content: center;
+    font-family: ${theme.fonts.pill.family};
+    font-weight: ${theme.fonts.pill.weight};
+    font-size: ${theme.fonts.pill.size}px;
+    line-height: ${theme.fonts.pill.lineHeight}px;
+    letter-spacing: ${theme.fonts.pill.letterSpacing}em;
+`}
+`;
+
+export const PillInput = styled.input`
+  /*
+Hide checkbox visually but remain accessible to screen readers.
+Source: https://polished.js.org/docs/#hidevisually
+*/
+  border: 0;
+  clip: rect(0 0 0 0);
+  clip-path: inset(50%);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+`;
+
+export const PillLabel = styled.span`
+  ${({ theme, isSelected }) => `
+  cursor: pointer;
+  margin: auto 0;
+  width: 100%;
+  display: flex;
+  padding: 4px;
+  padding-right: ${isSelected ? '4px' : '20px'};
+  padding-left: ${isSelected ? `${ACTIVE_CHOICE_ICON_SIZE}px` : '20px'};
+  align-items: center;
+  background-color: ${theme.colors.white};
+  color: ${theme.colors.gray700};
+  border: ${theme.borders.gray50};
+  border-radius: ${theme.button.borderRadius}px;
+
+  ${PillInput}:enabled:hover + & {
+    background-color: ${theme.colors.blueLight};
+  }
+
+  ${PillInput}:focus + & { 
+    border: ${theme.borders.action};
+  }
+
+  ${PillInput}:checked + & {
+    background-color: ${theme.colors.blueLight};
+  }
+
+  ${PillInput}:disabled + & {
+    opacity: 0.6;
+    cursor: default;
+  }
+`}
+`;
+PillLabel.propTypes = {
+  isSelected: PropTypes.bool,
+};
+
+export const FloatingTabLabel = styled(PillLabel)`
+  background-color: transparent;
+  padding: 4px 16px;
+  border-color: transparent;
+
+  ${PillInput}:checked + & {
+    box-shadow: ${({ theme }) => theme.floatingTab.shadow};
+    background-color: transparent;
+  }
+`;
+
+export const SwatchLabel = styled(PillLabel)`
+  ${({ hex }) => `
+    padding: 0;
+    align-items: center;
+    justify-content: center;
+    margin: 0 auto;
+    border-radius: 50%;
+    width: 26px;
+    height: 26px;
+    background-color: ${hex};
+
+    ${PillInput}:checked + &,
+    ${PillInput}:enabled:hover + & {
+      background-color: ${hex};
+      > svg {
+        opacity: 1.0;
+      }
+    }
+
+  `}
+`;
+SwatchLabel.propTypes = {
+  hex: PropTypes.string.isRequired,
+};
+
+export const ActiveSwatchIcon = styled(CheckmarkIcon).attrs(() => ({
+  width: ACTIVE_CHOICE_ICON_SIZE,
+  height: ACTIVE_CHOICE_ICON_SIZE,
+}))`
+  position: absolute;
+  opacity: 0;
+  margin: auto;
+  padding: 2px;
+  color: ${({ theme, hex = '#000' }) =>
+    hex.toLowerCase().includes('fff')
```

Comment:
This is a nice way to manage this. Should we get really light grays in the mix, we will likely have to calculate brightness/luminosity. But we can keep it simple for now! However since the `hex` property is `required`. We might not need the default fallback value. If we feel safer with that (and I do), then we don’t need to mark the prop as `required`.